Lib/CertUtils.php

```diff
@@ -0,0 +1,112 @@
+<?php
+App::uses('CakeLog', 'Log');
+
+class CertUtils {
+  /**
+   * Get Certificate Subject DN Environment Value
+   * Get Certificate Issuer DN Environment Value
+   *
+   * @param array Plugin Configuration
+   *
+   * @return string[]
+   * @example ['/CN /O ...', '/CN /O ...']
+   */
+  public static function getEnvValues($plg_cfg) {
+    // Get Certificate SDN and IDN mappings
+    $subject_dn_value = "";
+    $issuer_dn_value = "";
+    $subject_dn_map = !empty($plg_cfg['subject_dn_attribute']) ? $plg_cfg['subject_dn_attribute'] : null;
+    $issuer_dn_map = !empty($plg_cfg['issuer_dn_attribute']) ? $plg_cfg['issuer_dn_attribute'] : null;
+    // Get Subject DN env value if available
+    if(!empty($subject_dn_map)) {
+      $subject_dn_value = getenv($subject_dn_map);
+      $subject_dn_value = !empty($subject_dn_value) ? $subject_dn_value : "";
+    }
+    // Get Issuer DN env value if available
+    if(!empty($issuer_dn_map)) {
+      $issuer_dn_value = getenv($issuer_dn_map);
+      $issuer_dn_value = !empty($issuer_dn_value) ? $issuer_dn_value : "";
+    }
+
+    return array($subject_dn_value, $issuer_dn_value);
+  }
+
+  /**
+   * Get Certificate Subject DN Environment Value
+   * Get Certificate Issuer DN Environment Value
+   * 
+   * @param array Plugin Configuration
+   * @param array Attributes Cached from new OrgIdentity
+   *
+   * @return string[]
+   * @example ['/CN /O ...', '/CN /O ...']
+   */
+  public static function getEnvValuesFromCache($plg_cfg, $attribute_cache) {
+    // Get Certificate SDN and IDN mappings
+    $subject_dn_value = "";
+    $issuer_dn_value = "";
+    $subject_dn_map = !empty($plg_cfg['subject_dn_attribute']) ? $plg_cfg['subject_dn_attribute'] : null;
+    $issuer_dn_map = !empty($plg_cfg['issuer_dn_attribute']) ? $plg_cfg['issuer_dn_attribute'] : null;
+    // Get Subject DN env value if available
+    if(!empty($subject_dn_map)) {
+      $subject_dn_value = $attribute_cache[$subject_dn_map];
```

Comment:
is that equivalent to $subject_dn_value = !empty($attribute_cache[$subject_dn_map]) ? $attribute_cache[$subject_dn_map] : "";
?

Reply:
yes